pkg/yurthub/healthchecker/health_checker.go

```diff
@@ -138,63 +392,30 @@ func (c *checker) isHealthy() bool {
 	return c.clusterHealthy
 }
 
-func (c *checker) healthyCheckLoop(stopCh <-chan struct{}) {
-	intervalTicker := time.NewTicker(heartbeatFrequency)
-	defer intervalTicker.Stop()
-	healthyCnt := 0
-	isHealthy := false
-	var err error
+func (c *checker) setHealthy(healthy bool, t time.Time) {
 
-	for {
-		select {
-		case <-stopCh:
-			klog.Infof("exit normally in health check loop for %s", c.netAddress)
-			return
-		case <-intervalTicker.C:
-			for i := 0; i < c.failedRetry; i++ {
-				isHealthy, err = PingClusterHealthz(c.healthzClient, c.serverHealthzAddr)
-				if err != nil {
-					klog.V(2).Infof("ping cluster healthz with result, %v", err)
-					if !c.clusterHealthy {
-						// cluster is unhealthy, no need ping more times to check unhealthy
-						break
-					}
-				} else {
-					if c.clusterHealthy {
-						// cluster is healthy, no need ping more times to check healthy
-						break
-					}
-				}
-			}
-
-			if !isHealthy {
-				// cluster is unhealthy
-				healthyCnt = 0
-				if c.clusterHealthy {
-					c.Lock()
-					c.clusterHealthy = false
-					c.Unlock()
-					now := time.Now()
-					klog.Infof("cluster becomes unhealthy from %v, healthy status lasts %v", now, now.Sub(c.lastTime))
-					c.onFailureFunc(c.netAddress)
-					c.lastTime = now
-					metrics.Metrics.ObserveServerHealthy(c.netAddress, 0)
-				}
-			} else {
-				//  with continuous 2 times cluster healthy, unhealthy will changed to healthy
-				healthyCnt++
-				if !c.clusterHealthy && healthyCnt >= c.healthyThreshold {
-					c.Lock()
-					c.clusterHealthy = true
-					c.Unlock()
-					now := time.Now()
-					klog.Infof("cluster becomes healthy from %v, unhealthy status lasts %v", now, now.Sub(c.lastTime))
-					c.lastTime = now
-					metrics.Metrics.ObserveServerHealthy(c.netAddress, 1)
-				}
-			}
+	c.Lock()
+	defer c.Unlock()
+	c.clusterHealthy = healthy
+	c.lastTime = t
+	if !healthy {
+		if c.onFailureFunc != nil {
+			c.onFailureFunc(c.netAddress)
 		}
+		metrics.Metrics.ObserveServerHealthy(c.netAddress, 0)
+		return
 	}
+	metrics.Metrics.ObserveServerHealthy(c.netAddress, 1)
+}
+
+func (hcm *healthCheckerManager) genRestConfig(host string) *rest.Config {
+	cfg := &rest.Config{
+		Host:      "https://" + host,
+		Transport: hcm.tp.CurrentTransport(),
+		Timeout:   hcm.heartbeatTimeoutSeconds,
+	}
+	return cfg
+
 }
 
 func PingClusterHealthz(client *http.Client, addr string) (bool, error) {
```

Comment:
PingClusterHealthz() has not been used now. how about delete it?

Reply:
yurtctl uses this method to detect the health status of yurthub when setup yurthub.